packages/block-editor/src/store/selectors.js

```diff
@@ -463,6 +463,68 @@ export function getBlockHierarchyRootClientId( state, clientId ) {
 	return parent;
 }
 
+/**
+ * Given a block client ID, returns the hierarchy tree of client ID.
+ *
+ * @param {Object} state    Editor state.
+ * @param {string} clientId Block from which tree will be created.
+ *
+ * @return {Array} Hierarchy tree of client ID.
+ */
+export function getTree( state, clientId ) {
```

Comment:
Isn't this the same selector as `getBlockParents` but in reverse order?

Reply:
Basically yes but there are two difference:
1. It also include `clientId` at the beginning of the return Array
2. It use `createSelector` function which I believe brings some memoization here and when I tried to use it instead `getTree` it do not works as expected. I tried to go in that way:

```
export function getTree( state, clientId ) {
	return [ clientId, ...getBlockParents( state, clientId ).reverse() ];
}
```

Reply:
> when I tried to use it instead getTree it do not works as expected

Sorry, in which way it didn't work as expected?

Reply:
I'm checking this once again and I have noticed that there is another reason behind use `getTree`. When I wrote this function there wasn't `getBlockParents` implemented yet.

Back to your question:

>Sorry, in which way it didn't work as expected?

I replace `getTree` with the lines posted above and in some cases it returns `undefined` instead of top most `parentId` in `block.native.js` when `firstToSelectId` is called. 

However when I remove `createSelector` which wraps the `getBlockParents` selector body it works the same as my `getTree` implementation.

Reply:
I manage to refactor to extract common logic for those two functions. Please take a look in recently pushed commit if that's ok for you.